test/test_docker_api_deploy.py

```diff
@@ -43,129 +74,140 @@ def test_docker_api_deploy(self):
             f"http://{self.docker_environment.database_host}:{self.docker_environment.ports.bucketfs}/"
-            f"{bucket_name}/test-flavor-release-{release_name}.tar.gz",
+            f"{upload_path}",
         )
 
-        expected_path_in_bucket = (
-            bfs.path.build_path(
-                backend=bfs.path.StorageBackend.onprem,
-                url=f"http://{self.docker_environment.database_host}:{self.docker_environment.ports.bucketfs}",
-                bucket_name=bucket_name,
-                service_name=bucketfs_name,
-                username="w",
-                password=self.docker_environment.bucketfs_password,
-                verify=False,
-            )
-            / f"test-flavor-release-{release_name}.tar.gz"
+        expected_path_in_bucket = self._build_bfs_path(
+            bucket_name, bucketfs_name, expected_extension, path, release_name
         )
-
         # Compare UDF path of `bucket_path` until bfs.path.PathLike implements comparison
         self.assertEqual(
             expected_path_in_bucket.as_udf_path(),
             deploy_result.bucket_path.as_udf_path(),
         )
 
         self.validate_file_on_bucket_fs(
-            bucket_name, f"test-flavor-release-{release_name}.tar.gz"
+            bucket_name,
+            path,
+            release_name,
+            expected_extension,
+            compression=compression,
+        )
+
+    def validate_file_on_bucket_fs(
+        self,
+        bucket_name: str,
+        path: Optional[str],
+        release_name: str,
+        expected_extension: str,
+        compression: bool,
+    ):
+        host = self.docker_environment.database_host
+        port = self.docker_environment.ports.bucketfs
+        bucketfs_username = self.docker_environment.bucketfs_username
+        bucketfs_password = self.docker_environment.bucketfs_password
+        expected_file = self._expected_file(release_name, expected_extension)
+        path_in_bucket = "/".join(filter(None, [bucket_name, path, expected_file]))
+        with TemporaryDirectory() as tmpdir:
+            url = f"http://{bucketfs_username}:{bucketfs_password}@{host}:{port}/{path_in_bucket}"
+            file_name = f"{tmpdir}/{expected_file}"
+            cmd = [
+                "curl",
+                "--silent",
+                "--show-error",
+                "--fail",
+                url,
+                "--output",
+                file_name,
+            ]
+            p = subprocess.run(cmd, capture_output=True)
+            p.check_returncode()
+
+            # "r:gz" / "r:" makes 'tarfile.open' to raise an exception if file is not in requested format
+            tar_mode = "r:gz" if compression else "r:"
+            with tarfile.open(name=file_name, mode=tar_mode) as tf:  # type: ignore
+                tf_members = tf.getmembers()
+                last_tf_member = tf_members[-1]
+                assert last_tf_member.name == "exasol-manifest.json"
```

Comment:
Better implementation would be by using `python_bucketfs`, however this is currently blocked by https://github.com/exasol/bucketfs-python/issues/204